hazelcast/src/main/java/com/hazelcast/map/impl/record/RecordFactory.java

```diff
@@ -16,21 +16,16 @@
 
 package com.hazelcast.map.impl.record;
 
-import com.hazelcast.config.InMemoryFormat;
-import com.hazelcast.nio.serialization.Data;
-
 /**
- * TODO need a better name than RecordFactory!
+ * Factory for creating records. Created for every partition.
  *
  * @param <T> the type of object which is going to be created.
  */
 public interface RecordFactory<T> {
 
-    Record<T> newRecord(Data key, Object value);
+    Record<T> newRecord(Object value);
 
     void setValue(Record<T> record, Object value);
```

Comment:
These are methods that do not belong on a factory. A factory has a create method. Things like setValue and isEquals are not part of factory behavior. So what you have created here is something else than a factory.